### DISCUSSION TITLE
Best Way To Create A Flutter Package?

### DISCUSSION THREAD

**Discussion by mcmah309** [fzyzcjy/flutter_rust_bridge#2125] [Q&A] [answered]
What is the best way to create a flutter package/plugin/library? Is it copying https://github.com/fzyzcjy/flutter_rust_bridge/tree/master/frb_example/flutter_package as outlined here https://cjycode.com/flutter_rust_bridge/manual/integrate/cargokit#quickstart ? The only other entry in the docs I can find is this section https://cjycode.com/flutter_rust_bridge/manual/integrate/library which walks through manually setting everything up with is less desirable and likely prone to being outdated.

Is there an effort to add this to the cli - `flutter_rust_bridge_codegen create --plugin`?

**Comment by fzyzcjy** [comment]
Yes, currently I guess copy-and-paste-and-modify the frb_example/flutter_package may be helpful.

> Is there an effort to add this to the cli - flutter_rust_bridge_codegen create --plugin?

I was thinking there may not be a lot of people using this, thus I chose the more manual way. But maybe I am wrong, and feel free to PR for this! Alternatively, I may work on it later (but the priority may not be super high)

**Reply by mcmah309** [reply, marked as answer]
Added an issue and PR https://github.com/fzyzcjy/flutter_rust_bridge/pull/2144

**Comment by 0xNF** [comment]
I'm also interested in this. My use case involves loading multiple rust libraries into a flutter project, so integrating with just one rust repo per project isn't ideal. Flutter plugins seem like the best way to add multiple libraries at once, but the docs aren't super helpful, for instance, the quickstart page lists a `--template=plugin` option but it doesn't exist in the current release, and the included readme file is the default flutter one.

**Reply by fzyzcjy** [reply]
> the quickstart page lists a --template=plugin option but it doesn't exist in the current release

Hmm, it should be in 2.1.0 thanks to https://github.com/fzyzcjy/flutter_rust_bridge/pull/2144 @mcmah309 . Could you please check whether it is there?

**Reply by 0xNF** [reply]
ah, it might be in 2.1. I'm still using the version I `cargo install`''d from last week ,which I think is 2.0.
